django_nitro_mailer/tests/test_tasks.py

```diff
@@ -185,3 +185,47 @@ def test_send_mass_email_success(mock_send_mass_mail: MagicMock, mock_get_connec
     email_logs = EmailLog.objects.all()
     for email_log in email_logs:
         assert email_log.result == EmailLog.Results.SUCCESS
+
+
+@pytest.mark.django_db
+@patch("smtplib.SMTP")
+def test_sync_backend_sends_email(mock_send_messages: MagicMock) -> None:
+    settings.EMAIL_BACKEND = "django_nitro_mailer.backends.SyncBackend"
+    settings.NITRO_MAILER_EMAIL_BACKEND = "django.core.mail.backends.smtp.EmailBackend"
+
+    backend = SyncBackend()
+    email_message = EmailMessage(
+        subject="Test Subject",
+        body="This is a test message.",
+        from_email="from@example.com",
+        to=["to@example.com"],
+    )
+    mock_send_messages.return_value.sendmail.return_value = None
+    sent_count = backend.send_messages([email_message])
+
+    assert sent_count == 1
+    assert EmailLog.objects.count() == 1
+    mock_send_messages.assert_called_once()
+
+
+@pytest.mark.django_db
+@patch("smtplib.SMTP")
+def test_sync_backend_sends_email_failure(mock_smtp: MagicMock) -> None:
+    settings.EMAIL_BACKEND = "django_nitro_mailer.backends.SyncBackend"
+    settings.NITRO_MAILER_EMAIL_BACKEND = "django.core.mail.backends.smtp.EmailBackend"
+
+    backend = SyncBackend()
+    email_message = EmailMessage(
+        subject="Test Subject",
+        body="This is a test message.",
+        from_email="from@example.com",
+        to=["to@example.com"],
+    )
+
+    mock_smtp.return_value.sendmail.side_effect = Exception("Failed to send email")
+
+    sent_count = backend.send_messages([email_message])
+
+    assert sent_count == 0
+    assert EmailLog.objects.count() == 1
```

Comment:
Sprawdź też czy na pewno to jest `EmailLog` z `result = failure`; `EmailLog.objects.filter(result=fail).count()`